src/api/server.js

```diff
@@ -1,103 +1,120 @@
-import bodyParser from 'body-parser';
-import cookieParser from 'cookie-parser';
-import express from 'express';
-import https from 'https'
-import helmet from 'helmet';
-import fs from 'fs'
+import bodyParser from "body-parser"
+import cookieParser from "cookie-parser"
+import express from "express"
+import https from "https"
+import helmet from "helmet"
+import fs from "fs"
 import {
-  BOT_PORT,
-  WEB_APPLICATION,
-  REST_APPLICATION,
-  HTTPS_CHOICE,
-  WEBAPP_PORT,
-  WEBAPP_HOST,
-  SSL_CRT_LOCATION,
-  SSL_KEY_LOCATION,
-  // SSL_CA_LOCATION
-} from '../helpers/constants'
-import useWebAndRoutes from './webapi';
-import useRESTRoutes from './restapi';
+	BOT_PORT,
+	WEB_APPLICATION,
+	REST_APPLICATION,
+	HTTPS_CHOICE,
+	WEBAPP_PORT,
+	WEBAPP_HOST,
+	SSL_CRT_LOCATION,
+	SSL_KEY_LOCATION
+	// SSL_CA_LOCATION
+} from "../helpers/constants"
+import useWebAndRoutes from "./webapi"
+import useRESTRoutes from "./restapi"
 
-// set upload destination for attachments sent to broadcast with multer 
+// set upload destination for attachments sent to broadcast with multer
 const startServer = () => {
-  const app = express();
-  app.use(helmet()); //security http headers
+	const app = express()
+	app.use(helmet()) //security http headers
 
-  // parse application/x-www-form-urlencoded
-  app.use(bodyParser.urlencoded({ extended: false }));
-  // parse application/json
-  app.use(bodyParser.json());
-  app.use(cookieParser());
+	// parse application/x-www-form-urlencoded
+	app.use(bodyParser.urlencoded({ extended: false }))
+	// parse application/json
+	app.use(bodyParser.json())
+	app.use(cookieParser())
 
-  app.use((error, req, res, next) => {
+	app.use((error, req, res, next) => {
+		if (error instanceof SyntaxError) {
+			console.log("bodyParser:", error)
+			res.statusCode = 400
+			res.type("txt").send(error.toString())
+		} else {
+			next()
+		}
+	})
 
-    if (error instanceof SyntaxError) {
-      console.log('bodyParser:', error);
-      res.statusCode = 400;
-      res.type('txt').send(error.toString());
-    } else {
-      next();
-    }
-  });
+	// add cors for development
+	// TODO: set conditional for NODE_ENV to match and set the right origin host header - 8000 for dev, 4545 for prod
 
-  // add cors for development
-  // TODO: set conditional for NODE_ENV to match and set the right origin host header - 8000 for dev, 4545 for prod
+	// if webapplication or rest
+	if (WEB_APPLICATION.value == "yes" || REST_APPLICATION.value == "yes") {
```

Comment:
We have not done this check correctly.  There is a higher level TOKEN that needs to be set for either Web based application: WEB_INTERFACE
Need to verify that WEB_INTERFACE is "yes" before testing the WEB_APPLICATION or the REST_APPLICATION

It is possible that if a user re-configures the bot that WEB_INTERFACE is "no" but WEB_APPLICATION or REST_APPLICATION could be "yes"